pilot/pkg/model/service.go

```diff
@@ -833,6 +834,23 @@ func (u NoopAmbientIndexes) WorkloadsForWaypoint(scope WaypointScope) []*Workloa
 
 var _ AmbientIndexes = NoopAmbientIndexes{}
 
+type AddressInfo struct {
+	*workloadapi.Address
+	// Labels for the workload. Note these are only used internally, not sent over XDS
+	Labels map[string]string
```

Comment:
Labels on AddressInfo is a bit wonky. The labels are with the workload, not an address

Reply:
the current WorkloadInfo.Labels are used in `pilot/pkg/networking/core/v1alpha3/waypoint.go.findWaypointServices()`.  I wasn't sure if the new Service API objects would need to be returned by `pilot/pkg/model/push_context.go.WorkloadsForWaypoint()`.  In which case those labels would be lost once Workload is wrapped in the workloadapi.Address object

Reply:
as far as I can tell  findWaypointServices calls WorkloadsForWaypoint which returns a Workload, not an Address.

However, Idon't think we actually want the label selecting there; its pretty broken. Workload already maintains a list of services, we should instead use that as the source of truth